cli_tools/common/utils/logging/service/literal_loggable.go

```diff
@@ -17,30 +17,80 @@ package service
 type literalLoggable struct {
 	strings   map[string]string
 	int64s    map[string][]int64
+	bools     map[string]bool
 	traceLogs []string
 }
 
 func (w literalLoggable) GetValue(key string) string { return w.strings[key] }
 
+func (w literalLoggable) GetValueAsBool(key string) bool { return w.bools[key] }
+
 func (w literalLoggable) GetValueAsInt64Slice(key string) []int64 { return w.int64s[key] }
 
 func (w literalLoggable) ReadSerialPortLogs() []string { return w.traceLogs }
 
-// SingleImageImportLoggable returns a Loggable that is pre-initialized with the metadata
-// fields that are relevant when importing a single image file.
-func SingleImageImportLoggable(fileFormat string, sourceSize, resultSize int64, matchResult string, inflationTypeStr string, inflationTimeInt64 int64, shadowInflationTimeInt64 int64, traceLogs []string) Loggable {
+// SingleImageImportLoggableBuilder initializes and builds a Loggable with the metadata
+// fields that are relevant when importing a single image.
+type SingleImageImportLoggableBuilder struct {
+	fileFormat                string
+	sourceSize                int64
+	resultSize                int64
+	matchResult               string
+	inflationTypeStr          string
+	inflationTimeInt64        int64
+	shadowInflationTimeInt64  int64
+	isUEFICompatibleImageBool bool
+	isUEFIDetectedBool        bool
+	traceLogs                 []string
+}
+
+// SetDiskAttributes sets disk related attributes.
+func (b *SingleImageImportLoggableBuilder) SetDiskAttributes(fileFormat string, sourceSize int64,
+	resultSize int64, isUEFICompatibleImageBool bool, isUEFIDetectedBool bool) *SingleImageImportLoggableBuilder {
+
+	b.fileFormat = fileFormat
+	b.sourceSize = sourceSize
+	b.resultSize = resultSize
+	b.isUEFICompatibleImageBool = isUEFICompatibleImageBool
+	b.isUEFIDetectedBool = isUEFIDetectedBool
+	return b
+}
+
+// SetInflationAttributes sets inflation related attributes.
+func (b *SingleImageImportLoggableBuilder) SetInflationAttributes(matchResult string, inflationTypeStr string,
+	inflationTimeInt64 int64, shadowInflationTimeInt64 int64) *SingleImageImportLoggableBuilder {
+
+	b.matchResult = matchResult
+	b.inflationTypeStr = inflationTypeStr
+	b.inflationTimeInt64 = inflationTimeInt64
+	b.shadowInflationTimeInt64 = shadowInflationTimeInt64
+	return b
+}
```

Comment:
This is better than having one large func with 10+ args. 

Maybe a nit, but an even better improvement would be where builder functions would set each field directly into a loggable, instead of saving them in temporary fields within builder then copying them on Build() (Build() could just be basically a get() as it would return an already built loggable).

Something like:
b := service.NewSingleImageImportLoggableBuilder()

return b.WithMatchResult(matchResult).WithInflationType(inflationTypeStr)...WithFileFormat(fileFormat).WithUefiDetected(uefiDetected).Build()

where one With* func would be something like:

func (b *SingleImageImportLoggableBuilder) WithMatchResult(matchResult string) {
  b.loggable.strings[shadowDiskMatchResult] = matchResult
}

NewSingleImageImportLoggableBuilder() would have to initialise any maps etc. in b.loggable though.

You can even remove `With` prefix for brevity.

Reply:
Sounds good. Modified mostly as you suggested. Still grouped them other than having a With func for each field.